bundles/config/org.eclipse.smarthome.config.core/src/main/java/org/eclipse/smarthome/config/core/net/internal/NetworkConfigOptionProvider.java

```diff
@@ -47,6 +49,12 @@
                     .filter(a -> a.getAddress() instanceof Inet4Address);
             return ipv4Addresses.map(a -> new ParameterOption(a.toString(), a.toString())).collect(Collectors.toList());
         }
+
+        if (param.equals(PARAM_BROADCAST_ADDRESS)) {
+            List<String> broadcastAddrList = NetUtil.getAllBroadcastAddresses();
+            broadcastAddrList.add("255.255.255.255");
```

Comment:
I just stumbled across this address again and by looking up what it means on wikipedia I found:

> It is the broadcast address of the zero network or 0.0.0.0, which in Internet Protocol standards stands for this network, i.e. the local network. Transmission to this address is limited by definition, in that it is never forwarded by the routers connecting the local network to other networks.

So by setting this we would be back to "random" because "this network" is not really defined if you have multiple network interfaces, right? So I guess we should remove this one.

Reply:
It depends on the OS. Windows will broadcast to one interface. I believe that is true for all BSD derived systems, and windows got its TCP stack from BSD. My RPi broadcasts on all interfaces, and from what I know that would be the case for Linux. This has nothing to do with routing, as we are talking about multipe network interfaces on the same host. The standard does not define if each interface needs to broadcast in a multi-homed system, so it is implemented differently in different OS's. I added it because it allows an easy multipe interface broadcast on some systems (but sure not all). It is also the behaviour most bindings using broadcast to 255.255.255.255 currently have.
The better, OS independent, way is indeed not to use this. The default with this PR will be to use the broadcast address of the primary IP (with fallback to first if not defined). Secondly, selecting a value other than 255.255.255.255 here will broadcast to that specific interface. Then 255.255.255.255 is there as a last resort to support old behaviour with minimal change.
Ultimately, the best fix for broadcast to all interfaces would be to iterate over them. But that has a bigger impact on the bindings that use it than what this PR tries to do. @caffineehacker has a more complete solution in mind for this, adding methods to take care of the broadcasts and response callbacks. My PR was meant to be a minimum impact, but considerable improvement, making the behaviour more predictable, but leaving the possibility to still have the old behaviour.

Reply:
@mherwege Thank you for the clarification.